creusot/tests/should_succeed/type_invariants/generated.mlcfg

```diff
@@ -198,7 +196,367 @@ module Generated_UseFoo
+    type self = f,
+    type args = (),
+    predicate Precondition0.precondition = Precondition0.precondition,
+    predicate Inv0.inv = Inv0.inv,
+    predicate Inv1.inv = Inv2.inv,
+    predicate PostconditionOnce0.postcondition_once = PostconditionOnce0.postcondition_once,
+    predicate Inv2.inv = Inv1.inv,
+    type Output0.output = t
+  let rec cfg take_closure [#"../generated.rs" 26 0 26 51] [@cfg:stackify] [@cfg:subregion_analysis] (f : f) : t
+    requires {[#"../generated.rs" 26 41 26 42] Inv0.inv f}
+    ensures { [#"../generated.rs" 26 50 26 51] Inv1.inv result }
+    
+   = [@vc:do_not_keep_trace] [@vc:sp]
+  var _0 : t;
+  var f : f = f;
+  {
+    goto BB0
+  }
+  BB0 {
+    _0 <- ([#"../generated.rs" 27 4 27 7] CallOnce0.call_once f ());
+    f <- any f;
+    goto BB1
+  }
+  BB1 {
+    goto BB2
+  }
+  BB2 {
+    return _0
+  }
+  
+end
+module CreusotContracts_Resolve_Impl2_Resolve_Stub
+  type t
+  predicate resolve (self : t)
+end
+module CreusotContracts_Resolve_Impl2_Resolve_Interface
+  type t
+  predicate resolve (self : t)
+  val resolve (self : t) : bool
+    ensures { result = resolve self }
+    
+end
+module CreusotContracts_Resolve_Impl2_Resolve
+  type t
+  predicate resolve (self : t) =
+    [#"../../../../../creusot-contracts/src/resolve.rs" 36 8 36 12] true
+  val resolve (self : t) : bool
+    ensures { result = resolve self }
+    
+end
+module Generated_TestClosure_Closure0_Type
+  use Generated_Sum10_Type as Generated_Sum10_Type
+  type generated_testclosure_closure0  =
+    | Generated_TestClosure_Closure0 (Generated_Sum10_Type.t_sum10)
+    
+end
+module Generated_TestClosure_Closure0_Interface
+  use export Generated_TestClosure_Closure0_Type
+  use Generated_Sum10_Type as Generated_Sum10_Type
+  clone CreusotContracts_Invariant_Inv_Stub as Inv1 with
+    type t = Generated_Sum10_Type.t_sum10
+  clone CreusotContracts_Invariant_Inv_Stub as Inv0 with
+    type t = generated_testclosure_closure0
+  let function field_0 [#"../generated.rs" 32 12 32 19] (self : generated_testclosure_closure0) : Generated_Sum10_Type.t_sum10
+    
+   = [@vc:do_not_keep_trace] [@vc:sp]
+    [#"../generated.rs" 1 0 1 0] let Generated_TestClosure_Closure0 a = self in a
+  predicate resolve [#"../generated.rs" 32 12 32 19] (_1 : generated_testclosure_closure0) =
+    [#"../generated.rs" 1 0 1 0] true
+  predicate precondition [#"../generated.rs" 32 12 32 19] (self : generated_testclosure_closure0) (_ : ()) =
+    [#"../generated.rs" 1 0 1 0] true
+  predicate postcondition_once [#"../generated.rs" 32 12 32 19] (self : generated_testclosure_closure0) (_ : ()) (result : Generated_Sum10_Type.t_sum10)
+    
+   =
+    [#"../generated.rs" 1 0 1 0] true
+  val generated_TestClosure_Closure0 [#"../generated.rs" 32 12 32 19] (_1 : generated_testclosure_closure0) : Generated_Sum10_Type.t_sum10
+    requires {[#"../generated.rs" 1 0 1 0] Inv0.inv _1}
+    ensures { [#"../generated.rs" 32 12 32 19] Inv1.inv result }
```

Comment:
Does it even make sense to have a contract on the `generated_TestClosure_Closure0` function? I would expect it to only ever be called through the `Fn*` traits which have independent contracts...

Reply:
> edit: I was wrong 

yep, direct calls to the closure go through directly